samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -86,12 +76,8 @@ private String getHostName() {
     }
   }
 
-  public interface ZkLeaderElectorListener {
-    void onBecomingLeader();
-  }
-
   @Override
-  public boolean tryBecomeLeader() {
+  public boolean tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
```

Comment:
If you merely want to wait until you are the leader, an API could be 
`awaitLeadership()` that blocks until you are the leader.